### DISCUSSION TITLE
Readings in prep for week 4

### DISCUSSION THREAD

**Discussion by benmiller314** [benmiller314/dsam2023fall#6] [Reading responses etc]
Hi, all! We're in a (sort of) new space, but the procedure is the same: after reading the [introduction to _10 PRINT_](https://doi.org/10.7551/mitpress/9040.001.0001) (Montfort et al), [Zach Whalen's close reading of Bertram's _Travesty Generator_](http://digitalhumanities.org:8081/dhq/vol/17/2/000707/000707.html), the beginning of [Ruha Benjamin's _Race After Technology_](https://pitt.primo.exlibrisgroup.com/permalink/01PITT_INST/e8h8hp/alma9999863389206236), and [Tasneem Raja's article on coding and literacy](https://www.motherjones.com/media/2014/06/computer-science-programming-code-diversity-sexism-education/), head back here and let us know where your mind went: For example, you might...

* Highlight new ideas or key terms, especially those that you expect will be useful in thinking about digital projects.
* Highlight sentences, passages, or methods that struck you as especially well done, perhaps worth imitating.
* Link outward to something a passage reminds you of, explaining where you're leading us.
* Pose questions to your classmates about a passage, especially (but not exclusively) if something is confusing.
* Respond to any of the above – which you can now do directly, using threads! Joy.

Where possible, please try to point us to at least one particular passage from the text you're responding to; that will help to keep our discussion grounded.

**Comment by SidraNArshad** [comment]
Ruha Benjamin’s book Race After Technology includes a section titled “Tailoring: Targeting” that touches on the privacy issues we face on platforms such as Netflix save our search history and the previous media that we’ve consumed, and use this data to try and advertise other films and TV shows, a form of “tailored marketing” as Benjamin says. I think this points to a larger privacy issue that Benjamin touches on throughout this reading, specifically our own general passivity in regard to our information being used without our permission. This issue is not unique to any one platform as other such companies save and sell our private information to advertising agencies to better target us. We’ve just been desensitized to our general lack of online privacy, it’s become less significant to us, and I just want to know if this trend will continue into the future, in terms of how much further companies are going to go with it. We already know our phones are always tracking our locations, what comes next? I see people online say all the time that future generations are going to look back and be horrified. My only response to that would be how much worse will it be for them? Will privacy online just become an illusion? And the reality of the situation, that Benjamin touches on, is that such issues are not being confined to the digital sphere.

**Reply by Khushboobhutani** [reply]
Montfort’s article on BASIC teases us with the “social, political and aesthetic dimensions of code,” an idea that reaches realization in Ruha Benjamin’s _Race After Technology._ During one of our discussions in the previous classes, we had toyed with the idea of codes being affected by culture and vice versa. Benjamin elaborates on this interaction and argues how machines and algorithms can be racist. To me, this was an extremely engaging read because Benjamin affords a temporal dimension to codes and examines not only the emergence of a racist Internet but also traces the ways in which it is produced and sustained by technical sector and commercial products that are themselves shaped by historical inequalities. Her work critiques software that stereotypes and erases identity, difference and agency. She writes, “[t]he datasets and models used in these systems are the culmination of particular tools, people, and power structures that foreground one way of seeing or judging over another.” In doing so, the word code also acquires literary dimensions as it is deployed for people of a particular race. As we read technology through the lenses of history and experience, one is made aware of the dangers of tech, an issue that Sidra has also raised.

**Comment by chloe-cmu** [comment]
The readings this week made me think more about whether or not coding is a language. The answer, of course, is complicated, but I think I'm coming around to the idea that coding is a language like any other, but one with power enforcement built into it even more than human languages (which of course have many mechanisms for instituting hierarchy). Montford et al's exercise of breaking down each element of 10 Print and explaining what each chunk meant was illustrative, and on a certain level right in line with how one might learn a language. But it also brought up for me a way that coding is really very different from language, which is that coding contains no possibility of refusal, only of errors. Whalen's article provides an example of Bertram's work, where the artist intentionally explores this idea of error vs failure/system shutdown. (It sounds like Bertram's work is a really interesting innovation on the line break or ending line of poems from the perspective of computational poetry.) But in general, coding means accepting the starting parameters and going from there, without any opportunity to disagree or diverge. 

Benjamin's introduction makes the point that coding, without significant intervention, will only reflect and redouble existing inequalities, while seeming more objective and progressive than previous methods. (As a historian, I was happy to see Gibran Muhammad's *The Condemnation of Blackness* cited.) The Mother Jones piece made the vital point that we must understand computing as a way of thinking, and understand how a computer's "perspective" might be formulated and how it might approach problems. Combined, these two perspectives tell us that we will likely adopt computing as broad social mindset that allows us to handwave inequality away as a rational or preferable outcome, as we have many times before. 

Adopting the mindset of computing while framing it as a 'valueless' approach sets a dangerous precedent. If there can be no refusal, there can be no consent. This dynamic plays out in the way I think a lot of people (myself included) use the Internet and deal with privacy concerns. It's not really a possibility for people to opt out of using Internet services if they want to participate in public life. Sidra talks about this issue too. I agree with her generally, but I would argue that privacy online has already become an illusion. And I think that our passivity in regards to the way companies use our information is more than desensitization (though that's definitely part of it). It borders on learned helplessness (which I might not be using right). But since we live in a social system that has repeatedly demonstrated its unwillingness to meaninfully curb

**Comment by amj169** [comment]
I want to focus on Ruha Benjamin’s Race After Technology. This was the last piece that I read, and as I was taking notes up until this point, I was looking for something to make talking about code as literacy or access to programming feel more practical and less theoretical. I was struck by this because the New Jim Code felt like it did something different. It diagnosed systemic racism and capitalism as the root of inequity in the coding word (I don’t know what to call this field? I’ve typed and deleted like 4 different categorization). I was especially interested in the Raising Robots section because it dropped the curtain as far as how racial inequities continue to develop in online spaces. I have a sense that some people who are immersed in silicon valley may really think they are doing the good work because they aren’t intentionally playing a role in racist software development. Over the summer I attended a week long  automated writing workshop where different speakers would come and give lectures and workshops. I distinctly remember one who right off the bat identified himself as ‘not using tech or programming like’ because he was largely speaking to an Arts and Humanities crowd. If that weren’t enough, when people started asking questions and raising concerns about racial biases within AI (specifically the way in which some LLMs are being trained) he assured us that it wasn’t really that biased because it was scraping data from the internet that was equally available and already existing. This reminds me of what Benjamin says when he talks about intentionality in response to John McWhorter’s remark about intentionality being controlled by the individual. I’m not sure people recognize the degree in which racism and other harmful biases infiltrate these systems. This makes me wonder though if the New Jim Code is perhaps just as fantastical and far away as something like prison abolition, UBI, universal healthcare, or whatever other system capitalism delays or prevents entirely from getting off the ground.

**Comment by MustafaKandil** [comment]
In "Coding is the New Literacy," the author highlights the accessibility of coding education through online platforms like Codecademy, suggesting that anyone can acquire coding skills with interactive lessons. This democratization of coding education challenges the traditional view that coding is reserved for the tech-savvy elite. However, it also raises questions about the qualifications of instructors and the effectiveness of these methods.

The concept of "computational thinking" is introduced as a way of thinking with computers. It emphasizes the imaginative capacity to envision how digital information can be manipulated through computational techniques. Computational thinking involves breaking down complex tasks into smaller logical steps, akin to following a recipe. This perspective implies that thinking with computers is not solely about coding but rather about problem-solving and creative problem decomposition. It prompts us to consider whether one can think effectively without any knowledge of computers. While computational thinking is a valuable skill, traditional forms of thinking and problem-solving should not be undermined or overshadowed by it.

"Any Means Necessary to Refuse Erasure by Algorithm," explores how computational processes can convey the lived experiences shaped by colonialism, oppression, and injustice. The author suggests that algorithms and code can be powerful tools for expressing and challenging social realities. This reading underscores the idea that code is not just an abstract, technical language; it has social and political dimensions. It reminds us that code can be a medium for activism and resistance against erasure and injustice.

"10 Print Chr$(205.5+Rnd(1)); : Goto 10," delves into the concept of code as a cultural text. It emphasizes that code is not purely mathematical but is influenced by historical and social contexts. This reading advocates for a critical examination of code as a cultural resource. It suggests that code should be appreciated not only for its instrumental value but also for its role in shaping culture, aesthetics, and human-computer interactions.

It is evident that code is more than just a technical skill; it is a form of literacy with profound cultural, social, and political implications. To understand code critically, scholars should engage with programming, recognizing its relationship with culture and its capacity to reflect and influence societal values. Code is not cryptic or irrelevant but a meaningful and dynamic aspect of our digital world, deserving of attention and analysis. The readings invite us to consider the multifaceted nature of coding and computational thinking. They encourage us to recognize code as a valuable form of literacy and to explore its cultural and social significance. In an increasingly digitized world, understanding code is not only a practical skill but also a critical lens through which we can better comprehend our evolving society.
